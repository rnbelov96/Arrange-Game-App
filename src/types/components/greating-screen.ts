import { DimentionType } from '../general-types';

export type GreatingScreenPropsType = {
  dimention: DimentionType,
  onInputChange: (value: DimentionType) => void,
  onStart: () => void,
  isFinished: boolean,
}
