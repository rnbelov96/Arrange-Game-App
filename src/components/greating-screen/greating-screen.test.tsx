import React from 'react';
import renderer from 'react-test-renderer';
import { PureGreatingScreen } from './greating-screen';

describe('GreatingScreen', () => {
  test('should render screen with greating message', () => {
    const tree = renderer
      .create(
        <PureGreatingScreen
          dimention="three"
          isFinished={false}
          onInputChange={() => {}}
          onStart={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render screen with winning message', () => {
    const tree = renderer
      .create(
        <PureGreatingScreen
          dimention="three"
          isFinished
          onInputChange={() => {}}
          onStart={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
