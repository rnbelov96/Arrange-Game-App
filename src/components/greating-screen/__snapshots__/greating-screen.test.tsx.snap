// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GreatingScreen should render screen with greating message 1`] = `
<div
  className="sc-AykKC jOmOxD"
>
  <h1
    className="sc-AykKD iUedYT"
  >
    Arrange Game
  </h1>
  <h2
    className="sc-AykKE hJFghG"
  >
    Chose dimention:
  </h2>
  <select
    className="sc-AykKF dlkSEx"
    onChange={[Function]}
    value="three"
  >
    <option
      value="three"
    >
      three
    </option>
    <option
      value="four"
    >
      four
    </option>
    <option
      value="five"
    >
      five
    </option>
  </select>
  <button
    className="sc-AykKG hcNXkq"
    onClick={[Function]}
    type="button"
  >
    Start
  </button>
</div>
`;

exports[`GreatingScreen should render screen with winning message 1`] = `
<div
  className="sc-AykKC jOmOxD"
>
  <h1
    className="sc-AykKD iUedYT"
  >
    WINNER!!!
  </h1>
  <h2
    className="sc-AykKE hJFghG"
  >
    Chose dimention:
  </h2>
  <select
    className="sc-AykKF dlkSEx"
    onChange={[Function]}
    value="three"
  >
    <option
      value="three"
    >
      three
    </option>
    <option
      value="four"
    >
      four
    </option>
    <option
      value="five"
    >
      five
    </option>
  </select>
  <button
    className="sc-AykKG hcNXkq"
    onClick={[Function]}
    type="button"
  >
    Start
  </button>
</div>
`;
